### ISSUE TITLE
I can't get sudo to work with this at all.

### ISSUE THREAD

**Issue by halcyonardency** [alkersan/rundeck-openstack-plugin#2]
Do you have tips on how to get sudo to work?  It seems to fail on passing along the sudo password to the target host.   This doesn't happen for nodes I used outside of this plugin.

**Comment by alkersan** [comment]
There is really not much meta-information can be taken from Openstack about hosts. All this plugin does is just fetches list of hostnames and IPs from Openstack's Nova. And there is no way to tell how to login to each VM because openstack don't have such info.
As described in [rundeck manual](http://rundeck.org/docs/plugins-user-guide/ssh-plugins.html), there are 3 places where auth attributes can be set:
- Node level, but as described earlier - this plugin currently don't have facilities to fill-in such info
- Project level - if some attribute is not set on Node level - it can be looked up in project properties
- Framework level - just like project, but more global and works across projects

In our environment I used Project properties to set missing ssh parameters, such as username, keyfile, etc. Here is excerpt from project config:

```
#Fri Jan 01 12:53:54 EET 2016
#edit below
project.description=internal-dev-lab
project.name=internal-dev-lab
project.ssh-authentication=privateKey
project.ssh-key-storage-path=keys/id_rundeck
project.ssh.user=rundeck
resources.source.1.type=openstack
resources.source.1.config.id=os-dev-lab
resources.source.1.config.endpoint=*****
resources.source.1.config.tenant=*****
resources.source.1.config.username=rundeck
resources.source.1.config.password=*****
resources.source.1.config.refresh-interval=30
```

On project level all logins will be done from `rundeck` user with specified keyfile.
Regarding your problem with sudo I suggest first to try pull up [sudo related options](http://rundeck.org/docs/plugins-user-guide/ssh-plugins.html#secondary-sudo-password-authentication) to project level.

**Comment by halcyonardency** [comment]
Yeah, that's the thing.  I set the sudo options at the project level -- and it still didn't seem to work. It looks like you can turn sudo prompt wait stuff on, but it never passes a password across (or uses the right password).  For whatever reason, it seems unique to this module (so weird).  I might hack in the module to set some default sudo options per each node config, to see if that fixes it up.

**Comment by alkersan** [comment]
@xyntrix what other node sources do you use? If you give some more info about your environment, I'll try to reproduce similar one and look what's wrong with sudo password. In our lab we have simple setup with passwordless sudoers only.
